### ISSUE TITLE
PyCodeStyle does not remember setting (off)

### ISSUE THREAD

**Issue by johnsturgeon** [mmshivesh/Python-Nova.novaextension#8]
Every time I restart Nova, PyCodeStyle shows me errors.  To fix it, I have to go into the Python-Nova settings and toggle it 'on' and 'off' again.

**Comment by Gugu7264** [comment]
I agree, it's very disturbing. You can also change a setting of PyCodeStyle so it does apply.

**Comment by mmshivesh** [comment]
Hmm, i'll take a look at this, i've been really busy these past few weeks and i will come back to fixing bugs soon

**Comment by samdoran** [comment]
I run into this issue daily as well.

**Comment by samdoran** [comment]
It seems like this may be related to the fact that settings are not getting loaded on initial launch.
https://github.com/mmshivesh/Python-Nova.novaextension/blob/edce8a6abb6fcdec6223b93f5f198e316a8e1a1f/Scripts/main.js#L331


Changing any setting after the extension has loaded causes the correct settings to take affect. Odd because this is same method call.

https://github.com/mmshivesh/Python-Nova.novaextension/blob/edce8a6abb6fcdec6223b93f5f198e316a8e1a1f/Scripts/main.js#L267

**Comment by mmshivesh** [comment]
@samdoran You are correct. The whole preferences loading thing seems to be very wonky. I'm primarily a python dev and I've barely used JS, but if anyone has more insights/better ways to parse/handle preferences, I'm all ears.

**Comment by johnsturgeon** [comment]
Sad to say.  Because I love Nova. I gave in and moved to PyCharm for Python.  Still use nova but for python it's not ready for prime time

**Comment by mmshivesh** [comment]
IMO, the bigger problem I'm grappling with is the lack of great open source Python Language servers. The current LS that this extension uses is Palantir's Python LS but it's slow. Microsoft per usual hasn't open sourced their excellent PyLance server that's bundled with VSCode. Their older Python-Language-Server hasn't received updates since last year.

I might be able to get everything working if I write substantially more code, doing stuff akin to what Nova's [typescript](https://github.com/apexskier/nova-typescript) extension does.

**Comment by samdoran** [comment]
> I'm primarily a python dev and I've barely used JS before this project

Same here. I know enough JS to be dangerous, but not very useful. 😁

> IMO, the bigger problem I'm grappling with is the lack of great open source Python Language servers.

I see what you mean. It's unfortunate that PyLance is not open source. There is a [fork of python language server](https://github.com/python-lsp/python-lsp-server) that seems actively maintained. No idea if it's any faster than Palanitr's.

**Comment by mmshivesh** [comment]
> There is a fork of python language server that seems actively maintained. No idea if it's any faster than Palantir's.

Yes, I did take a look at that and am using it rn. But it's essentially the same, (afaik, it's a drop in replacement for Palantir's version). It's nothing groundbreaking like Microsoft's Pylance tbh.

**Comment by guillaumealgis** [comment]
Probably fixed in https://github.com/mmshivesh/Python-Nova.novaextension/pull/19

Maybe also related to https://github.com/mmshivesh/Python-Nova.novaextension/pull/18
